src/standard/utils.html

```diff
@@ -370,47 +370,57 @@
       var link = document.head.querySelector('[href="' + href + '"]');
       var self = this;
 
-      if (link == null) {
-        link = document.createElement('link');
-        link.rel = 'import';
-        link.__dynamicImport = true;
-      }
-
-      if (optAsync) {
-        link.setAttribute('async', '');
-      }
-
       var loadListener = function(event) {
-        link.removeEventListener('load', loadListener);
-        link.removeEventListener('error', errorListener);
+        cleanup();
         // In case of a successful load, cache the load event on the link so
         // that it can be used to short-circuit this method in the future when
         // it is called with the same href param.
-        link.__dynamicImportLoaded = true;
 
+        link.__loaded = true;
+        link.__error = false;
         if (onload != null) {
           return onload.call(self, event);
         }
       };
+
       var errorListener = function(event) {
-        link.removeEventListener('load', loadListener);
-        link.removeEventListener('error', errorListener);
+        cleanup();
         // In case of an error, remove the link from the document so that it
         // will be automatically created again the next time `importHref` is
         // called.
-        document.head.removeChild(link);
+        link.__loaded = true;
+        link.__error = true;
```

Comment:
The intention of the change is to discard this `link`. What is the benefit of tracking state on the `link` if it is about to be discarded?